File: example/java-example/src/main/java/com/java/example/model/BaseResponse.java
package com.java.example.model;


import com.example.shared.model.UserData;
import lombok.Builder;

import java.time.LocalDate;
import java.time.LocalDateTime;
import java.time.LocalTime;

@Builder
public record BaseResponse(
        Boolean success,
        Long timestamp,
        String message,
        LocalDate date,
        LocalDateTime dateTime,
        LocalTime time,
        UserData userData) {

}

File: example/kotlin-example/src/main/kotlin/com/kotlin/example/model/BaseResponse.kt
package com.kotlin.example.model

import com.example.shared.model.UserData
import java.time.LocalDate
import java.time.LocalDateTime
import java.time.LocalTime


data class BaseResponse(
    val success: Boolean,
    val timestamp: Long,
    val message: String,
    val time: LocalTime,
    val date: LocalDate,
    val dateTime: LocalDateTime,
    val userData: UserData
)
